hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveViewCommits.java

```diff
@@ -0,0 +1,442 @@
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(NS)
+            .withQuery("hive", "select * from ns.tbl")
+            .create();
+    viewLocation = new Path(view.location());
+  }
+
+  @AfterEach
+  public void dropTestView() throws IOException {
+    viewLocation.getFileSystem(HIVE_METASTORE_EXTENSION.hiveConf()).delete(viewLocation, true);
+    catalog.dropView(VIEW_IDENTIFIER);
+  }
+
+  @Test
+  public void testSuppressUnlockExceptions() {
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    ViewMetadata metadataV1 = ops.current();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k1", "v1").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1).containsEntry("k1", "v1");
+
+    HiveViewOperations spyOps = spy(ops);
+
+    AtomicReference<HiveLock> lockRef = new AtomicReference<>();
+
+    when(spyOps.lockObject())
+        .thenAnswer(
+            i -> {
+              HiveLock lock = (HiveLock) i.callRealMethod();
+              lockRef.set(lock);
+              return lock;
+            });
+
+    try {
+      spyOps.commit(metadataV2, metadataV1);
+      HiveLock spyLock = spy(lockRef.get());
+      doThrow(new RuntimeException()).when(spyLock).unlock();
+    } finally {
+      lockRef.get().unlock();
+    }
+
+    ops.refresh();
+
+    // the commit must succeed
+    assertThat(ops.current().properties()).hasSize(0);
+  }
+
+  /**
+   * Pretends we throw an error while persisting, and not found with check state, commit state
+   * should be treated as unknown, because in reality the persisting may still succeed, just not yet
+   * by the time of checking.
+   */
+  @Test
+  public void testThriftExceptionUnknownStateIfNotInHistoryFailureOnCommit()
+      throws TException, InterruptedException {
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k1", "v1").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1).containsEntry("k1", "v1");
+
+    HiveViewOperations spyOps = spy(ops);
+
+    failCommitAndThrowException(spyOps);
+
+    ops.refresh();
+
+    assertThat(ops.current()).as("Current metadata should not have changed").isEqualTo(metadataV2);
+    assertThat(metadataFileExists(metadataV2)).as("Current metadata should still exist").isTrue();
+    assertThat(metadataFileCount(ops.current()))
+        .as(
+            "New metadata files should still exist, new location not in history but"
+                + " the commit may still succeed")
+        .isEqualTo(2);
+  }
+
+  /** Pretends we throw an error while persisting that actually does commit serverside. */
+  @Test
+  public void testThriftExceptionSuccessOnCommit() throws TException, InterruptedException {
```

Comment:
please update this one to 
```
@Test
  public void testThriftExceptionSuccessOnCommit() throws TException, InterruptedException {
    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();

    ViewMetadata metadataV1 = ops.current();
    assertThat(metadataV1.properties()).hasSize(0);

    view.updateProperties().set("k1", "v1").commit();
    ops.refresh();
    ViewMetadata metadataV2 = ops.current();
    assertThat(metadataV2.properties()).hasSize(1).containsEntry("k1", "v1");

    HiveViewOperations spyOps = spy(ops);

    // Simulate a communication error after a successful commit
    commitAndThrowException(ops, spyOps);
    // Shouldn't throw because the commit succeeds even though an exception is thrown
    spyOps.commit(metadataV2, metadataV1);

    assertThat(ops.current()).as("Current metadata should have not changed").isEqualTo(metadataV2);
    assertThat(metadataFileExists(ops.current()))
        .as("Current metadata file should still exist")
        .isTrue();
  }
```